py/builtinimport.c

```diff
@@ -367,9 +367,10 @@ STATIC mp_obj_t process_import_at_level(qstr full_mod_name, qstr level_mod_name,
         // formerly known as "weak links").
         #if MICROPY_MODULE_WEAK_LINKS
         if (stat == MP_IMPORT_STAT_NO_EXIST && module_obj == MP_OBJ_NULL) {
-            char *umodule_buf = vstr_str(path);
+            char umodule_buf[MICROPY_ALLOC_PATH_MAX];
```

Comment:
`path` is a vstr that is allocated to PATH_MAX already and free to use. I don't think there's any need (stack-)allocate another buffer.

I think all that needs to change here is strcpy to strncpy, either using `MICROPY_ALLOC_PATH_MAX` or `path->alloc`, but rather than doing that it would be neater to do avoid the use of str(n)cpy altogether:

```c
assert(path->len == 0);
vstr_add_char(path, 'u');
vstr_add_str(path, qstr_str(level_mod_name));
qstr umodule_name = qstr_from_str(vstr_str(path)); // or qstr_from_strn(path->buf, path->len)
```

(the vstr will automatically bounds check the fixed buffer in vstr_add_str).

Reply:
I'm not convinced that `path->len` will always be 0 and don't want to introduce an unexpected `assert()` failure so I'm just using `vstr_reset()` first.